coap-gateway/service/signIn.go

```diff
@@ -138,7 +139,20 @@ func signInPostHandler(req *mux.Message, client *Client, signIn CoapSignInReq) {
 	}
 
 	if newDevice {
-		deviceSubscriber, err := grpcgwClient.NewDeviceSubscriber(req.Context, signIn.DeviceID, client.server.rdClient, client.server.resourceSubscriber)
+		deviceSubscriber, err := grpcgwClient.NewDeviceSubscriber(req.Context, signIn.DeviceID, func() func() (when time.Time, err error) {
+			var count uint64
+			maxRand := client.server.config.APIs.COAP.KeepAlive.Timeout / 2
+			if maxRand <= 0 {
```

Comment:
what is rand ?

Reply:
`rand(n)` generate random number between `0` and `n-1` - to avoid reconnect in same time.